server/middlewares/user.js

```diff
@@ -3,333 +3,317 @@ import db from '../models/incident';
+      return res.status(400).json({
+        status: 400,
+        message: 'Password should be 8 to 20 characters long'
+      });
+    }
+
+    if (confirmPassword === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Password cannot be undefined'
+      });
+    }
+
+    if (confirmPassword !== password) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Password do not match'
+      });
+    }
+
+    if (confirmPassword === '') {
+      return res.status(400).json({
+        status: 400,
+        message: 'Password field cannot be empty'
+      });
+    }
+
+    if (confirmPassword.length < 8 || confirmPassword.length > 20) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Password should be 8 to 20 characters long'
+      });
+    }
+
+    if (phoneNumber === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'phoneNumber cannot be undefine'
+      });
+    }
+
+    if (typeof phoneNumber !== 'string') {
+      return res.status(400).json({
+        status: 400,
+        message: 'PhoneNumber should be a string'
+      });
+    }
+
+    if (phoneNumber === '') {
+      return res.status(400).json({
+        status: 400,
+        message: 'phoneNumber field cannot be empty'
+      });
+    }
+
+    if (phoneNumber.length < 10 || phoneNumber.length > 12) {
+      return res.status(400).json({
+        status: 400,
+        message: 'phoneNumber should be 10 to 12 long'
+      });
+    }
+
+    const phoneNumberVerifier = /^[\d]+$/;
+    if (!phoneNumberVerifier) {
+      res.status(400).json({
+        status: 400,
+        message: 'Invalid PhoneNumber'
+      });
+    }
+    if (phoneNumberVerifier.length < 7 || phoneNumberVerifier.length > 12) {
+      res.status(400).json({
+        status: 400,
+        message: 'phoneNumber length too short '
+      });
+    }
+
+    if (username === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'username cannot be undefined'
+      });
+    }
+
+    if (username === '') {
+      return res.status(400).json({
+        status: 400,
+        message: 'username cannot be empty'
+      });
+    }
+    if (typeof username !== 'string') {
+      return res.status(400).json({
+        status: 400,
+        message: 'username must be a character',
+      });
+    }
+    const foundUserName = db.userDb.find(user => user.username === username);
+    if (foundUserName) {
+      return res.status(409).json({
+        status: 409,
+        message: 'UserName already exists!'
+      });
+    }
+
+    req.body.firstName = firstName;
+    req.body.lastName = lastName;
+    req.body.otherNames = otherNames;
+    req.body.email = email;
+    req.body.password = password;
+    req.body.confirmPassword = confirmPassword;
+    req.body.phoneNumber = phoneNumber;
+    req.body.username = username;
```

Comment:
I don't understand your assignment to `req.body`

Reply:
Acknowledged and fixed